src/internal/storedPixelDataToCanvasImageDataPseudocolorLUT.js

```diff
@@ -44,7 +57,20 @@ function storedPixelDataToCanvasImageDataPseudocolorLUT (image, grayscaleLut, co
     }
   } else {
     while (storedPixelDataIndex < numPixels) {
-      grayscale = grayscaleLut[pixelData[storedPixelDataIndex++]];
+      //  Find the grayscale of 4 channels image data by calculating relative luminance value
+      if (image.color && ((image.width * image.height * 4) === numPixels)) {
```

Comment:
You shouldn't do this type of if statement inside the while loop, it's going to be too slow. Do the if statement outside the loop, and if we need to follow a path where we need to calculate luminance, use a different path and while loop to do that instead, rather than checking this expression many many thousands of times

Reply:
I have updated to do separate path for relative luminance calculation in case of 4 channel color image and removed the conditions inside the while loop.